### PR TITLE
Begins support for image-filters

### PR DESCRIPTION
**Pull request by fdansv**
Ref. #79 

We're beginning to add image-filters like those found in Mapnik/Carto. For now, adding colorize-alpha.

cc @javisantana

### CODE REVIEW

**Comment by fdansv** [comment]
Also fixes #100

File: lib/torque/renderer/simpleheat.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        this._data.push(point);
+        return this;
+    },
+
+    clear: function () {
+        this._data = [];
+        return this;
+    },
+
+    radius: function (r, blur) {
```
Comment (javisantana) [inline lib/torque/renderer/simpleheat.js]: I think these methods are not used

File: lib/torque/renderer/point.js [hunk 2]
```diff
@@ -321,6 +342,31 @@ var carto = global.carto || require('carto');
...
+        var gradient = {};
+        var increment = 1/colorize.length;
+        for (var i = 0; i < colorize.length; i++){
+          var key = increment * i + increment;
+          var rgb = colorize[i].rgb;
+          var formattedColor = rgbToHex(rgb[0], rgb[1], rgb[2]);
+          gradient[key] = formattedColor;
+        }
+        this._filters.gradient(gradient);
+        this._filters.draw(0.1);
```
Comment (javisantana) [inline lib/torque/renderer/point.js]: if the param is not used, remove it

File: lib/torque/renderer/point.js [hunk 3]
```diff
@@ -2,6 +2,7 @@ var torque = require('../');
 var cartocss = require('./cartocss_render');
 var Profiler = require('../profiler');
 var carto = global.carto || require('carto');
+var filters = require('./simpleheat');
```
Comment (javisantana) [inline lib/torque/renderer/point.js]: change name if this is not going to be only for heat

File: lib/torque/renderer/point.js [hunk 4]
```diff
@@ -148,13 +159,14 @@ var carto = global.carto || require('carto');
...
       }
+      
       return canvas;
     },
 
     //
     // renders all the layers (and frames for each layer) from cartocss
     //
-    renderTile: function(tile, key) {
+    renderTile: function(tile, key, pos) {
```
Comment (javisantana) [inline lib/torque/renderer/point.js]: is pos needed?

File: lib/torque/renderer/point.js [hunk 5]
```diff
@@ -321,6 +342,31 @@ var carto = global.carto || require('carto');
...
+        var colorize = this._style['image-filters'].args;
+        var gradient = {};
+        var increment = 1/colorize.length;
+        for (var i = 0; i < colorize.length; i++){
+          var key = increment * i + increment;
+          var rgb = colorize[i].rgb;
+          var formattedColor = rgbToHex(rgb[0], rgb[1], rgb[2]);
+          gradient[key] = formattedColor;
+        }
+        this._filters.gradient(gradient);
```
Comment (javisantana) [inline lib/torque/renderer/point.js]: cache grandient generation

**Comment by fdansv** [comment]
@javisantana all changes from your comments applied!

File: lib/torque/renderer/torque_filters.js [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+        ctx.fillStyle = gradient;
+        ctx.fillRect(0, 0, 1, 256);
+
+        this._grad = ctx.getImageData(0, 0, 1, 256).data;
+
+        return this;
+    },
+
+    draw: function () {
+        if (!this._circle) {
```
Comment (javisantana) [inline lib/torque/renderer/torque_filters.js]: this is not needed i think

**Comment by fdansv** [comment]
:ok_hand:

**Pull request merged**